src/components/Common/Drawings/DrawingTab.tsx

```diff
@@ -0,0 +1,237 @@
+          />
+        </div>
+      ) : elements.length === 0 ? (
+        <div className="flex flex-col items-center justify-center text-gray-400">
+          <CareIcon icon="l-image" className="text-2xl mb-1" />
+          <span className="text-xs">{t("empty_drawing")}</span>
+        </div>
+      ) : (
+        <div
+          ref={svgContainerRef}
+          className="h-full w-full flex items-center justify-center p-2"
+        />
+      )}
+    </div>
+  );
+});
+
+// Add display name for debugging purposes
+ExcalidrawPreview.displayName = "ExcalidrawPreview";
+
+export const DrawingTab = (props: DrawingsTabProps) => {
+  const { qParams, updateQuery, Pagination, resultsPerPage } = useFilters({
+    limit: 15,
+    cacheBlacklist: ["name"],
+  });
+
+  const associatingId =
+    props.type === "encounter" ? props.encounter?.id : props.patientId;
+
+  const { data, isLoading } = useQuery({
+    queryKey: ["drawings", associatingId, qParams, resultsPerPage],
+    queryFn: query.debounced(metaArtifactApi.list, {
+      queryParams: {
+        object_type: "drawing",
+        associating_type: props.type,
+        name: qParams.name,
+        associating_id: associatingId,
+        limit: resultsPerPage,
+        offset: (qParams.page - 1) * resultsPerPage,
+      },
+    }),
+  });
+
+  return (
+    <div className="p-4">
+      <div className="flex items-center justify-between mb-4 gap-2">
+        <Input
+          id="search-by-name"
+          name="name"
+          placeholder={t("search_drawings")}
+          value={qParams.name}
+          onChange={(e) => updateQuery({ name: e.target.value })}
+          className="w-full sm:w-1/3"
+        />
+        <Button variant="white" onClick={() => navigate("drawings/new")}>
+          <CareIcon icon="l-pen" />
+          {t("new_drawing")}
+        </Button>
+      </div>
+      {isLoading ? (
+        <Loading />
+      ) : (
+        <>
+          {data?.results.length === 0 ? (
+            <div className="flex flex-col items-center justify-center p-8 text-gray-500">
+              <CareIcon icon="l-image" className="text-4xl mb-2" />
+              <p className="text-lg font-medium">{t("no_drawings_so_far")}</p>
+              <p className="text-sm">{t("create_new_drawing_message")}</p>
+            </div>
+          ) : (
+            <div className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-4">
+              {data?.results.map((drawing) => (
+                <Card
+                  key={drawing.id}
+                  className="overflow-hidden hover:shadow-md transition-shadow duration-200 group cursor-pointer"
+                  onClick={() => {
+                    if (props.type === "encounter") {
+                      navigate(
+                        `/facility/${props.encounter?.facility.id}/patient/${props.patientId}/encounter/${props.encounter?.id}/drawings/${drawing.id}`,
+                      );
+                    } else {
```

Comment:
we no longer need an if check here right? this wouldn't work for org. use cases right?